libs/api/ebpf_api.cpp

```diff
@@ -1593,7 +1593,7 @@ _initialize_ebpf_object_native(
     ebpf_assert(map_handles);
     ebpf_assert(program_handles);
 
-    object.native_module_handle = native_module_handle;
+    object.native_module_fd = _create_file_descriptor_for_handle(native_module_handle);
```

Comment:
`_create_file_descriptor_for_handle()` can fail, so we need to handle the error case also.

Alternatively, we can convert the handle to fd inside `_load_native_module()` itself after the IOCTL call returns. That will simplify the cleanup logic in case of failure.